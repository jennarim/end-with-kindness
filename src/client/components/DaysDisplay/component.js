import React from 'react';
import PropTypes from 'prop-types';

export default class DaysDisplay extends React.Component {
    render() {
        return (
            <div className="day-tracker">
                <div className="day-number"> {this.props.numberOfDays} </div>
                DAYS OF CONSECUTIVE KINDNESS
            </div>
        );
    }
}

DaysDisplay.propTypes = {
    numberOfDays: PropTypes.number.isRequired
};