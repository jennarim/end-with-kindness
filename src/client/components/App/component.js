import React from 'react';
import axios from 'axios';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { summary } from 'date-streaks';
import styled, { keyframes } from 'styled-components';
import { fadeIn, fadeInDown } from 'react-animations';

import ActList from './../ActList/component.js';
import DaysDisplay from './../DaysDisplay/component.js';
import ActForm from './../ActForm/component.js';
import About from './../About/component.js';
import Ideas from './../Ideas/component.js';
import Rules from './../Rules/component.js';
import ActPage from './../ActPage/component.js';
import NavigationBar from './../NavigationBar/component.js';

const FadeInDown = styled.div`animation: 1s ${keyframes`${fadeInDown}`}`;
const FadeIn = styled.div`animation: 1s ${keyframes`${fadeIn}`}`;

export default class App extends React.Component {
    state = {
        // count: 0,
        acts: []
    }

    componentDidMount() {
        axios.get("api/acts")
            .then(res => {
                console.log("axios:", res);
                const acts = res.data;

                // Get by most recent
                acts.reverse();
                this.setState({ acts });

                // Reorder for left-to-right masonry order
                // const reordered = this.reorder(this.state.acts, 3);
                // this.setState({ acts: reordered });
            });
    }

    reorder = (arr, columns) => {
        /*
            Credit to: https://github.com/jessekorzan/masonry-css-js
        */
        const cols = columns;
        const output = [];
        let col = 0;
        while (col < cols) {
            for (let i = 0; i < arr.length; i += cols) {
                let _val = arr[i + col];
                if (_val !== undefined)
                    output.push(_val);
            }
            col++;
        }
        return output;
    }

    addNewAct = newAct => {
        this.setState(prevState => ({
            acts: [newAct, ...prevState.acts]
        }));
    }

    getConsecutiveDays() {
        const dates = [...this.state.acts].map(act => act.datePosted);
        console.log(summary({ dates }));
        return summary({ dates }).currentStreak;
    }

    render() {
        return (
            <div>
                <Router>
                    <FadeInDown>
                        <NavigationBar />
                    </FadeInDown>
                    <Switch>
                        <Route exact path="/">
                            <div>
                                <DaysDisplay numberOfDays={this.getConsecutiveDays()} />

                                <FadeIn>
                                    <div className="post-container">
                                        <ActForm onSubmit={this.addNewAct} />

                                        <ActList acts={this.state.acts} />

                                    </div>
                                </FadeIn>


                                {/* <button onClick={() => this.setState({ count: this.state.count + 1 })}>{this.state.count}</button> */}
                            </div>
                        </Route>
                        <Route path="/ideas">
                            <Ideas />
                        </Route>
                        <Route path="/rules">
                            <Rules />
                        </Route>
                        <Route path="/about">
                            <About />
                        </Route>
                        <Route path="/act/:slug">
                            <ActPage />
                        </Route>
                    </Switch>
                </Router>
            </div >
        );
    }
}
