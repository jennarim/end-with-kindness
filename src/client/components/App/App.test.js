import React from "react";
import { shallow, mount } from "enzyme";
import App from "./component";
import toJson from "enzyme-to-json";
import DaysDisplay from "../DaysDisplay/component";

// Unit test - Render component in isolation
it("renders without crashing", () => {
    shallow(<App />);
});

// Integration test - Render component in its context
it("renders number of consecutive days", () => {
    const wrapper = mount(<App />);
    expect(wrapper.find(DaysDisplay)).toHaveLength(1);
});

it("renders correctly", () => {
    const tree = shallow(<App />);
    expect(toJson(tree)).toMatchSnapshot();
});

