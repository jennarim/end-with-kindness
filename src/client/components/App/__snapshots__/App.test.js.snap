// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <BrowserRouter>
    <styled.div>
      <NavigationBar />
    </styled.div>
    <Switch>
      <Route
        exact={true}
        path="/"
      >
        <div>
          <DaysDisplay
            numberOfDays={0}
          />
          <styled.div>
            <div
              className="post-container"
            >
              <ActForm
                onSubmit={[Function]}
              />
              <ActList
                acts={Array []}
              />
            </div>
          </styled.div>
        </div>
      </Route>
      <Route
        path="/ideas"
      >
        <Ideas />
      </Route>
      <Route
        path="/rules"
      >
        <Rules />
      </Route>
      <Route
        path="/about"
      >
        <About />
      </Route>
      <Route
        path="/act/:slug"
      >
        <withRouter(ActPage) />
      </Route>
    </Switch>
  </BrowserRouter>
</div>
`;
